services/director-v2/tests/integration/02/test_dynamic_sidecar_nodeports_integration.py

```diff
@@ -0,0 +1,874 @@
+
+    logger.warning("Patching to: DYNAMIC_SIDECAR_IMAGE=%s", image_name)
+    monkeypatch.setenv("DYNAMIC_SIDECAR_IMAGE", image_name)
+    monkeypatch.setenv("TRAEFIK_SIMCORE_ZONE", "test_traefik_zone")
+    monkeypatch.setenv("SWARM_STACK_NAME", "test_swarm_name")
+
+    monkeypatch.setenv("SC_BOOT_MODE", "production")
+    monkeypatch.setenv("DYNAMIC_SIDECAR_EXPOSE_PORT", "true")
+    monkeypatch.setenv("SIMCORE_SERVICES_NETWORK_NAME", network_name)
+    monkeypatch.delenv("DYNAMIC_SIDECAR_MOUNT_PATH_DEV", raising=False)
+    monkeypatch.setenv("DIRECTOR_V2_DYNAMIC_SCHEDULER_ENABLED", "true")
+    monkeypatch.setenv("DIRECTOR_V2_CELERY_SCHEDULER_ENABLED", "false")
+
+    settings = AppSettings.create_from_envs()
+
+    app = init_app(settings)
+    return app
+
+
+@pytest.fixture
+async def director_v2_client(
+    loop: BaseEventLoop, fast_api_app: FastAPI
+) -> httpx.AsyncClient:
+    async with LifespanManager(fast_api_app):
+        async with httpx.AsyncClient(
+            app=fast_api_app, base_url="http://testserver/v2"
+        ) as client:
+            yield client
+
+
+@pytest.fixture
+def client(fast_api_app: FastAPI) -> TestClient:
+    """required to avoid rewriting existing code"""
+    return TestClient(fast_api_app, raise_server_exceptions=True)
+
+
+@pytest.fixture
+async def cleanup_services_and_networks(
+    workbench_dynamic_services: Dict[str, Node],
+    current_study: ProjectAtDB,
+    director_v2_client: httpx.AsyncClient,
+) -> Iterator[None]:
+    yield None
+    # ensure service cleanup when done testing
+    async with aiodocker.Docker() as docker_client:
+        service_names = {x["Spec"]["Name"] for x in await docker_client.services.list()}
+
+        # grep the names of the services
+        for node_uuid in workbench_dynamic_services:
+            for service_name in service_names:
+                # if node_uuid is present in the service name it needs to be removed
+                if node_uuid in service_name:
+                    delete_result = await docker_client.services.delete(service_name)
+                    assert delete_result is True
+
+        project_id = f"{current_study.uuid}"
+
+        # pylint: disable=protected-access
+        scheduler_interval = (
+            director_v2_client._transport.app.state.settings.DYNAMIC_SERVICES.DYNAMIC_SCHEDULER.DIRECTOR_V2_DYNAMIC_SCHEDULER_INTERVAL_SECONDS
+        )
+        # sleep enough to ensure the observation cycle properly stopped the service
+        await asyncio.sleep(2 * scheduler_interval)
+        await ensure_network_cleanup(docker_client, project_id)
+
+
+@pytest.fixture
+def temp_dir(tmpdir: LocalPath) -> Path:
+    return Path(tmpdir)
+
+
+# UTILS
+
+
+async def _get_mapped_nodeports_values(
+    user_id: int, project_id: str, workbench: Workbench, db_manager: DBManager
+) -> Dict[str, InputsOutputs]:
+    result: Dict[str, InputsOutputs] = {}
+
+    # TODO: SAN please have a look into the following.
```

Comment:
@sanderegg please jave a look at the below comment and let's discuss it when you have time.